includes/REST_API/Stories_Lock_Controller.php

```diff
@@ -0,0 +1,212 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\REST_API;
+
+use WP_REST_Server;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_Error;
+use WP_Post_Type;
+
+/**
+ * API endpoint for post locking.
+ *
+ * Class Stories_Lock_Controller
+ */
+class Stories_Lock_Controller extends Stories_Base_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'web-stories/v1';
+		$this->rest_base = 'post-lock/(?P<id>\d+)';
+	}
+
+	/**
+	 * Registers routes for post locking.
+	 *
+	 * @return void
+	 * @see register_rest_route()
+	 *
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'lock_check' ],
+					'permission_callback' => [ $this, 'permissions_check' ],
+					'args'                => [
+						'id' => [
+							'description'       => __( 'Web Story ID.', 'web-stories' ),
+							'required'          => true,
+							'type'              => 'integer',
+							'sanitize_callback' => 'absint',
+						],
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Lock check, return true for now.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 *
+	 */
+	public function lock_check( $request ) {
+		return rest_ensure_response( $this->get_lock_status( $request->get_param( 'id' ) ) );
+	}
+
+	/**
+	 * Checks if current user can alter post locking.
+	 *
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 *
+	 */
+	public function permissions_check() {
+		if ( ! current_user_can( 'edit_web-stories' ) ) {
+			return new WP_Error( 'rest_forbidden', __( 'Sorry, you are not allowed.', 'web-stories' ), [ 'status' => rest_authorization_required_code() ] );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Taken from wp_refresh_post_lock().
+	 *
+	 * @param int $id Web Story ID.
+	 */
+	protected function get_lock_status( int $post_id ) {
+		$response = [];
+		$send     = [];
+		if ( ! $post_id ) {
+			return $response;
+		}
+
+		if ( ! current_user_can( 'edit_post', $post_id ) ) {
+			return $response;
+		}
+
+		$user_id = $this->check_post_lock( $post_id );
+		$user    = get_userdata( $user_id );
+		if ( $user ) {
+			$error = array(
+				/* translators: %s: User's display name. */
+				'text' => sprintf( __( '%s has taken over and is currently editing.' ), $user->display_name ),
+			);
+
+			if ( get_option( 'show_avatars' ) ) {
+				$error['avatar_src']    = get_avatar_url( $user->ID, array( 'size' => 64 ) );
+				$error['avatar_src_2x'] = get_avatar_url( $user->ID, array( 'size' => 128 ) );
+			}
+
+			$send['lock_error'] = $error;
+		} else {
+			$new_lock = $this->set_post_lock( $post_id );
```

Comment:
This could just use `wp_set_post_lock()`. No need to duplicate this code from core.

Reply:
Is that function available via the REST API?

Reply:
Ah, good point.

`wp-admin/includes/post.php` would need to be loaded in the REST API.

But we cannot just copy the core code for license reasons.